README.md

```diff
@@ -9,6 +9,13 @@ This proposal is a [stage 1 proposal](https://github.com/tc39/proposals/blob/mas
 
 Finding an element in an array is a very common programming pattern. 
 
+**The proposal has two major concerns:**
+
+1. Semantical. Which means `clearly representing the operation i want`.
+2. Performance. Avoid obvious overhead. And improve the constant factors in the time complexity or average-case complexity.

```

Comment:
generally speaking, performance isn’t reliable to reason about. v8, for example, is able to inline most array methods to be just as fast as a for loop - so without implementers claiming perf gains, it may not be reasonable to claim them here.

Reply:
There's two part of performance:
1. Overhead. Not sure about the v8 and others engine. But relative to direct find. There's a reverse/copy at least. Could engines avoid that operation too?
2. Time complexity. Even the worst-case complexity is O(N). But as I sad. It's may (Okay, may) improve the constant factors or average-case complexity if you know in some scenarios. 

These worthy of our attention IMO. So I think it's reasonable.

Reply:
O(2N) is identical to O(N), so it’s really just the first one.

Reply:
Yep. the `worst-case complexity` is both O(N). But the `constant factors` (2 to 1 in the case) will caused nearly 100% speed up.

Reply:
sure, but for small n (small being, not millions) - the common case - it doesn’t matter.

Reply:
And in some scenario. for example If we know the element is 80% chance in the second half. It's also caused some (Sorry I haven't calculate) speed up by `constant factors` improvement.

Reply:
> sure, but for small n (small being, not millions) - the common case - it doesn’t matter.

Well. I agreed...

Reply:
But I still prefer to mention it (not major). Because it would help in some performance sensitive scenarios. for example react render.

Reply:
> generally speaking, performance isn’t reliable to reason about

Is this a productive concern for the README of this repo @ljharb? If engine implementers or users find that this is not the case, they're free to bring this up in how we communicate this to readers.

Reply:
@DanielRosenwasser i'd say that "performance" is never a productive thing for a proposal to emphasize without concrete signals from implementers that it has an impact.

Reply:
I have updated the statement and says the performance improve as a sugar. PTAL again...